third_party/amd/lib/TritonAMDGPUTransforms/Utility.h

```diff
@@ -0,0 +1,24 @@
+#ifndef TRITON_THIRD_PARTY_AMD_LIB_TRITONAMDGPUTRANSFORMS_UTILITY_H_
+#define TRITON_THIRD_PARTY_AMD_LIB_TRITONAMDGPUTRANSFORMS_UTILITY_H_
+
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/Value.h"
+
+#include <functional>
+#include <limits>
+
+using namespace mlir;
+
+// DFS the def chain of 'value' starting from 'consumer' and will return the
+// minimum found when accumulating countFunc(op) for all non control flow ops
+// between value and the consumer. This function will traverse through for loop
+// iterations and to the outside of the loop to find all its producers.
+//    CountOp(Operation*) should return the value to accumulate for the
+//    operation
+// Returns 0 if there is an error traversing the def chain
+int findMinPathCountInDefChain(Value value, Operation *consumerOp,
+                               const std::function<int(Operation *)> &countFunc,
+                               int pathSum = 0,
+                               int foundMin = std::numeric_limits<int>::max());
```

Comment:
This should be something internal? I think you want this entry function to drop these but initialize local variables for them and call an internal function with these parameters?